YoCode/FileChangeChecker.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace YoCode
+{
+    public class FileChangeChecker
+    {
+        public static bool ProjectIsModified(Directory dir)
+        {
+            if (dir.OriginalPaths.Count != dir.ModifiedPaths.Count)
+            {
+                return true;
+            }
+            else {
+                for (int i = 0; i < dir.ModifiedPaths.Count; i++)
+                {
+                    FileStream ofs = File.Create(dir.OriginalPaths[i]);
+                    FileStream mfs = File.Create(dir.ModifiedPaths[i]);
```

Comment:
These static calls to `File.Create` will make this function impossible to test. I would make this a method on `IDirectory` that returns `Stream`, so you can mock it to return a `MemoryStream` like you do in the other test.